File: 27. Remove Element/27.js
// 27. Remove Element solution in JavaScript
// https://leetcode.com/problems/remove-element/description/

/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
 var removeElement = function(nums, val) {
    let length = nums.length;
    let i = 0;

    for (let j = 0; j < length; j++) {
        if (nums[j] != val) {
            nums[i] = nums[j];
            i++;
        }
    }

    return i;
};

File: 27. Remove Element/27.go
// 27. Remove Element solution in GoLang
// https://leetcode.com/problems/remove-element/description/

package main

func removeElement(nums []int, val int) int {
	length := len(nums)
	i := 0

	for j := 0; j < length; j++ {
		if nums[j] != val {
			nums[i] = nums[j]
			i++
		}
	}

	return i
}
